pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/blocks/TransferableBlockUtils.java

```diff
@@ -76,9 +76,13 @@ public static boolean isNoOpBlock(TransferableBlock transferableBlock) {
   public static Iterator<TransferableBlock> splitBlock(TransferableBlock block, DataBlock.Type type, int maxBlockSize) {
     List<TransferableBlock> blockChunks = new ArrayList<>();
     if (type == DataBlock.Type.ROW) {
-      // Use estimated row size, this estimate is not accurate and is used to estimate numRowsPerChunk only.
-      int estimatedRowSizeInBytes = block.getDataSchema().getColumnNames().length * MEDIAN_COLUMN_SIZE_BYTES;
-      int numRowsPerChunk = maxBlockSize / estimatedRowSizeInBytes;
+      // Use 1 row per chunk when block.getDataSchema().size() is 0, we may want to fine tune it.
+      int numRowsPerChunk = 1;
+      if (block.getDataSchema().size() != 0) {
+        // Use estimated row size, this estimate is not accurate and is used to estimate numRowsPerChunk only.
+        int estimatedRowSizeInBytes = block.getDataSchema().size() * MEDIAN_COLUMN_SIZE_BYTES;
+        numRowsPerChunk = maxBlockSize / estimatedRowSizeInBytes;
+      }
       Preconditions.checkState(numRowsPerChunk > 0, "row size too large for query engine to handle, abort!");
```

Comment:
nit: you may also want to adjust this precondition, as it's no longer helpful?

Reply:
I had an offline discussion with @walterddr . He  mentioned that 
1/ we should throw exception when dataSchema size is 0
2/ the current built query plan for `select 1 as one from {table} where 1 < 2` is not correct, we should update the query plan building logic.

Reply:
yeah dataSchema should not be zero once reach this far down. it looks like a bug somewhere else in the planner

Reply:
@walterddr, For select 1 from tbl. the dataSchema should int I guess?